src/Pagination/index.jsx

```diff
@@ -0,0 +1,414 @@
+          disabled={isLastPage}
+        />
+      </li>
+    );
+  }
+
+  renderScreenReaderSection() {
+    const { currentPage } = this.state;
+    const { buttonLabels, pageCount } = this.props;
+    return (
+      <div
+        className="sr-only"
+        aria-live="polite"
+        aria-relevant="text"
+        aria-atomic
+      >
+        {
+          getTextFromElement((
+            <div>
+              {buttonLabels.page}
+              {` ${currentPage}, `}
+              {buttonLabels.currentPage}
+              {', '}
+              {buttonLabels.pageOfCount}
+              {` ${pageCount}`}
+            </div>
+          ))
+        }
+      </div>
+    );
+  }
+
+  /*
+   * Uses recursion to generate an array of the buttons to display (i.e., page/ellipsis
+   * buttons) given the currently selected page, the max number of buttons to display, and
+   * the total number of pages.
+   */
+  renderPageButtons(range) {
+    const { currentPage } = this.state;
+    const { pageCount, maxPagesDisplayed } = this.props;
+
+    const pageRange = range || [...Array(pageCount).keys()].map(page => page + 1);
+    const currentIndex = pageRange.indexOf(currentPage);
+    const middleIndex = parseInt(pageRange.length / 2, 10) - 1;
+    const lastIndex = pageRange.length - 1;
+    const hasLeftEllipsis = pageRange[1] - pageRange[0] > 1;
+    const hasRightEllipsis = pageRange[lastIndex] - pageRange[lastIndex - 1] > 1;
+
+    let ellipsisCount = 0;
+
+    ellipsisCount += hasLeftEllipsis;
+    ellipsisCount += hasRightEllipsis;
+
+    if (pageRange.length <= maxPagesDisplayed - ellipsisCount) {
+      return pageRange.map((page) => {
+        const pageButtons = [this.renderPageButton(page)];
+
+        if (page === 1 && hasLeftEllipsis) {
+          pageButtons.push(this.renderEllipsisButton());
+        } else if (page === pageCount && hasRightEllipsis) {
+          pageButtons.unshift(this.renderEllipsisButton());
+        }
+
+        return pageButtons;
+      });
+    }
+
+    if (currentIndex > middleIndex) {
+      const first = pageRange.shift();
+      pageRange.shift();
+      pageRange.unshift(first);
+    } else {
+      const last = pageRange.pop();
+      pageRange.pop();
+      pageRange.push(last);
+    }
+
+    return this.renderPageButtons(pageRange);
+  }
+
+  render() {
+    return (
+      <nav
+        aria-label={this.props.paginationLabel}
+        className={this.props.className}
+      >
+        {this.renderScreenReaderSection()}
+        <ul className={styles.pagination}>
+          {this.renderPreviousButton()}
+          <ExtraSmall>
+            {this.renderPageOfCountButton()}
+          </ExtraSmall>
+          <LargerThanExtraSmall>
+            {this.renderPageButtons()}
+          </LargerThanExtraSmall>
+          {this.renderNextButton()}
+        </ul>
+      </nav>
+    );
+  }
+}
+
+const maxPagesDisplayedValidator = (props, propName, componentName) => {
```

Comment:
I noticed you were using the `airbnb-proptypes` in the ListBox PR, and meant to revisit this PR to start using the `between` validation now that we use it. I think I was just trying to avoid installing `airbnb-proptypes` for this one use case. But makes perfect sense to use it now that its used elsewhere!